processmanager/ebpf/ebpf.go

```diff
@@ -4,35 +4,16 @@
 package ebpf // import "go.opentelemetry.io/ebpf-profiler/processmanager/ebpf"
 
 import (
-	"context"
-	"errors"
 	"fmt"
-	"math/bits"
-	"reflect"
-	"sync"
-	"unsafe"
 
-	cebpf "github.com/cilium/ebpf"
-	"github.com/cilium/ebpf/features"
-	log "github.com/sirupsen/logrus"
 	"go.opentelemetry.io/ebpf-profiler/host"
 	"go.opentelemetry.io/ebpf-profiler/interpreter"
 	"go.opentelemetry.io/ebpf-profiler/libpf"
 	"go.opentelemetry.io/ebpf-profiler/lpm"
 	"go.opentelemetry.io/ebpf-profiler/metrics"
-	sdtypes "go.opentelemetry.io/ebpf-profiler/nativeunwind/stackdeltatypes"
-	"go.opentelemetry.io/ebpf-profiler/rlimit"
+	"go.opentelemetry.io/ebpf-profiler/nativeunwind/stackdeltatypes"
 	"go.opentelemetry.io/ebpf-profiler/support"
 	"go.opentelemetry.io/ebpf-profiler/util"
-	"golang.org/x/exp/constraints"
-)
-
-const (
-	// updatePoolWorkers decides how many background workers we spawn to
-	// process map-in-map updates.
-	updatePoolWorkers = 16
-	// updatePoolQueueCap decides the work queue capacity of each worker.
-	updatePoolQueueCap = 8
 )
 
 // EbpfHandler provides the functionality to interact with eBPF maps.
```

Comment:
If I understand correctly, this and rest of this PR is about separating `EbpfHandler` interface out from the `processmanager/ebpf` so that the interface can be included elsewhere without pulling in the actual ebpf code which does not compile?

Could we do this the otherway around? Just move this interface to some other package like `processmanager/ebpfapi`? Need to also check how this kind of split is done elsewhere.

Reply:
> If I understand correctly, this and rest of this PR is about separating EbpfHandler interface out from the processmanager/ebpf so that the interface can be included elsewhere without pulling in the actual ebpf code which does not compile?

Yes, this is correct in regards the changes in the `pmebpf` package.
There are some other changes in the `process` package, but they are slightly different - solving the compilation issues with build tags. I do think it makes sense to split process into `processapi` in a similar way you suggested `ebpfapi` - let me know if you think we should do this within the same change or as a separate or at all.

Reply:
> Could we do this the otherway around? Just move this interface to some other package like processmanager/ebpfapi? Need to also check how this kind of split is done elsewhere.

Yes, this should work. Applied